Tests/ZIPFoundationTests/ZIPFoundationEntryTests.swift

```diff
@@ -201,4 +201,10 @@ extension ZIPFoundationTests {
         XCTAssertEqual(dataDescriptorWithoutSignature?.uncompressedSize, 10)
         XCTAssertEqual(dataDescriptorWithoutSignature?.compressedSize, 10)
     }
+
+    func testEntryIsCompressed() throws {
+        let archive = self.archive(for: #function, mode: .read)
+        XCTAssert(archive["compressed"]?.isCompressed == true)
+        XCTAssert(archive["uncompressed"]?.isCompressed == false)
+    }
```

Comment:
The code for creating this test archive was:
```swift
var archiveURL = ZIPFoundationTests.resourceDirectoryURL
archiveURL.appendPathComponent(#function.replacingOccurrences(of: "()", with: ""))
archiveURL.appendPathExtension("zip")
let newArchive = Archive(url: archiveURL, accessMode: .create)!

let data = Data("This is just some test data that might be compressed or not".utf8)
try newArchive.addEntry(with: "uncompressed", type: .file, uncompressedSize: Int64(data.count), compressionMethod: .none, provider: { position, size in
    let position = Int(position)
    return data[position ..< (position + size)]
})
try newArchive.addEntry(with: "compressed", type: .file, uncompressedSize: Int64(data.count), compressionMethod: .deflate, provider: { position, size in
    let position = Int(position)
    return data[position ..< (position + size)]
})
```

Reply:
Looks good - thanks for providing a test case covering the addition!